Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs

```diff
@@ -395,6 +389,26 @@ public override void Enable()
             }
         }
 
+        private void CreateOrUpdateControllerForInteractionSource(InteractionSourceState interactionSourceState)
+        {
+            // If this is a new detected controller, raise source detected event with input system
+            // check needs to be here because GetController adds it to the activeControllers Dictionary
+            // this could be cleaned up because that's not clear
+            bool raiseSourceDetected = !activeControllers.ContainsKey(interactionSourceState.source.id);
+
+            var controller = GetController(interactionSourceState.source);
+
+            if (controller != null)
+            {
+                controller.UpdateController(interactionSourceState);
```

Comment:
I think we may want to swap the order of `UpdateController` and `RaiseSourceDetected`, like in the old `InteractionSourceDetected` implementation.

Otherwise, an app may receive button press events before a source detected event.

Reply:
done. In addition, I changed the names of two methods into GetOrAddController